pkg/api/error.go

```diff
@@ -17,67 +17,11 @@
 
 package api
 
-import (
-	"bytes"
-	"context"
-	"encoding/json"
-	"errors"
-	"fmt"
-	"io/ioutil"
-	"net/http"
-	"reflect"
-	"unsafe"
+import "github.com/elastic/cloud-sdk-go/pkg/api/apierror"
 
-	"github.com/go-openapi/runtime"
-)
-
-// UnwrapError unpacks an error message returned from autogenerated sdk.
+// UnwrapError (DEPRECATED) unpacks an error message returned from a client API
```

Comment:
I think the linter is upset about the use of `(DEPRECATED)` here 😄

Reply:
Addressed your comments and made some more changes in  bc6022f